kotest-framework/kotest-framework-engine/src/commonMain/kotlin/io/kotest/engine/test/interceptors/TimeoutInterceptor.kt

```diff
@@ -23,28 +38,79 @@ internal class TimeoutInterceptor(
       scope: TestScope,
       test: suspend (TestCase, TestScope) -> TestResult
    ): TestResult {
+      val timeout = testCase.config.timeout
 
-      // this timeout applies to the test itself. If the test has multiple invocations then
+      // This timeout applies to the test itself. If the test has multiple invocations, then
       // this timeout applies across all invocations. In other words, if a test has invocations = 3,
-      // each test takes 300ms, and a timeout of 800ms, this would fail, becauase 3 x 300 > 800.
-      logger.log { Pair(testCase.name.testName, "Switching context to add timeout ${testCase.config.timeout}") }
-
-      return when (val timeout = testCase.config.timeout) {
-         null -> test(testCase, scope)
-         else -> try {
-            withTimeout(timeout) {
-               test(testCase, scope.withCoroutineContext(coroutineContext))
-            }
-         } catch (t: Throwable) {
+      // each test takes 300ms, and a timeout of 800ms, this would fail, because 3 x 300 > 800.
+      logger.log { Pair(testCase.name.testName, "Switching context to add timeout $timeout") }
```

Comment:
Nit: do we need a `Pair` to construct a text message to log?

Reply:
`Logger.log` takes `() -> Pair<String?, String>` as a parameter, so I think yes